### PR TITLE
JS models unit tests

### PR DESCRIPTION
**Pull request by ascott1**
This adds unit tests to the application's models. @KimberlyMunoz and I worked closely on setting these.

## Additions
 
- Tests for each of the models (with the exception of `reg-model`)
  
## Changes
 
- Bumps the node dependency to 0.12.0
- Moves the definition model to its own module (it was in the initialize method of one of the views)
 
## Review
 
- @KimberlyMunoz 
 
## Preview

screen shot 2015-03-26 at 4 55 04 pm

### CODE REVIEW

File: regulations/static/regulations/js/source/models/meta-model.js [hunk 1]
```diff
@@ -26,7 +26,7 @@ var MetaModel = Backbone.Model.extend({
         // in the case of reg-model
         // content = markup to string representations of each reg paragraph/entity
         // loaded into the browser (rendered or not)
-        this.structure = this.structure || [];
+        this.structure = this.strutcture || [];
```
Comment (KimberlyMunoz) [inline regulations/static/regulations/js/source/models/meta-model.js]: Whoops. Looks like this typo is mine from messing up an ⌘-T.

File: regulations/static/regulations/js/unittests/specs/models/definition-model-spec.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    });
+
+    beforeEach(function(){
+        Resources.versionElements = {
+            toc: $('<nav id="toc" data-toc-version="2014-20681"></nav>'),
+        };
+    });
+
+    it('getAJAXUrl returns the correct URL endpoint with /definition supplemental path', function() {
+
```
Comment (KimberlyMunoz) [inline regulations/static/regulations/js/unittests/specs/models/definition-model-spec.js]: Missing assertion here.

**Comment by ascott1** [comment]
Updated to reflect @KimberlyMunoz's comments. Good catches!

**Comment by KimberlyMunoz** [comment]
Oops. Need a quick conflict fix.

**Comment by ascott1** [comment]
Merge conflict should be fixed!

**Comment by contolini** [comment]
So. much. green. :dollar: :moneybag: :dollar: :moneybag: :dollar: :moneybag:

**Comment by imuchnik** [comment]
@ascott1 Tests with no asserts are unethical :no_good:

**Comment by ascott1** [comment]
> @ascott1 Tests with no asserts are unethical 

And sometimes completely accidental :stuck_out_tongue_winking_eye:

**Pull request merged**